### PR TITLE
fix: BaseForm will show all error messages when multiple

### PR DESCRIPTION
**Pull request by sherizada-okta**
## Description:

Before:
Screen Shot 2022-01-24 at 6 45 21 PM

After:
Screen Shot 2022-01-24 at 6 44 15 PM

Before:
Screen Shot 2022-01-25 at 1 48 18 PM
After:
Screen Shot 2022-01-25 at 1 46 21 PM
## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-463004](https://oktainc.atlassian.net/browse/OKTA-463004)

### CODE REVIEW

File: src/v2/view-builder/internals/BaseForm.js [hunk 1]
```diff
@@ -130,16 +131,19 @@ export default Form.extend({
...
+          title = '',
+          message,
+        } = obj;
+        if(!msgClass || msgClass === INFO_MESSAGE_CLASS) {
           // add message as plain text
-          this.add(`<p>${obj.message}</p>`, '.ion-messages-container');
+          this.add(`<p>${message}</p>`, '.ion-messages-container');
         } else {
-          // add error callout with custom options
-          options = Object.assign(_.pick(obj, 'message', 'class'), options);
```
Comment (pradeepdewda-okta) [inline src/v2/view-builder/internals/BaseForm.js:138]: can you check after removing this.. custom error callouts still work ?

Comment (pradeepdewda-okta) [reply src/v2/view-builder/internals/BaseForm.js:138]: there are few views that provides `options` as arguments when they trigger a call to `showMessages`  when `showCustomFormErrorCallout` is not null..

File: playground/mocks/data/idp/idx/error-identify-multiple-errors.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
+{
+  "version": "1.0.0",
+  "stateHandle": "02f9EONFMnQZtAQHl7Gf4Ye-R4mXc8cOhBUSMc7ubQ",
+  "expiresAt": "2021-11-02T22:10:41.000Z",
+  "step":"IDENTIFY",
+  "intent":"LOGIN",
+  "remediation":{
```
Comment (pradeepdewda-okta) [inline playground/mocks/data/idp/idx/error-identify-multiple-errors.json:7]: i would also add another file that shows terminal view with multiple error messages.. meaning that it doesn't have `remediation` object instead only have `messages`

File: src/v2/view-builder/internals/BaseForm.js (hunk 1 above)
Comment (sherizada-okta) [reply src/v2/view-builder/internals/BaseForm.js:138]: How do I test custom error callouts?

File: test/testcafe/spec/Identify_spec.js [hunk 3]
```diff
@@ -397,3 +403,14 @@ test.requestHooks(identifyRequestLogger, baseIdentifyMock)('should show "Keep me
   doesCheckboxExist = identityPage.identifierFieldExists('.custom-checkbox [name="rememberMe"');
   await t.expect(doesCheckboxExist).eql(true);
 });
+
+test.requestHooks(identifyRequestLogger, errorsIdentifyMock)('should render each error message when there are multiple', async t => {
```
Comment (pradeepdewda-okta) [inline test/testcafe/spec/Identify_spec.js:407]: add a minor test for custom error as well..

File: test/testcafe/spec/Identify_spec.js [hunk 4]
```diff
@@ -397,3 +403,14 @@ test.requestHooks(identifyRequestLogger, baseIdentifyMock)('should show "Keep me
...
+  const identityPage = await setup(t);
+
+  const errors = identityPage.form.getAllErrorBoxTexts();
+  await t.expect(errors).eql([
```
Comment (pradeepdewda-okta) [inline test/testcafe/spec/Identify_spec.js:411]: also check for `Error` or `Info` class..

**Review by pradeepdewda-okta — APPROVED** [review]

**Pull request closed without merging**